contracts/protocols/polygon/uniswap/helpers.sol

```diff
@@ -248,26 +248,52 @@ abstract contract Helpers is DSMath {
             (token0, token1) = mintParams.tokenA < mintParams.tokenB
                 ? (mintParams.tokenA, mintParams.tokenB)
                 : (mintParams.tokenB, mintParams.tokenA);
-        }
-
-        // compute the liquidity amount
-        {
-            liquidity = LiquidityAmounts.getLiquidityForAmounts(
-                TickMath.getSqrtRatioAtTick(newCurrentTick),
-                TickMath.getSqrtRatioAtTick(mintParams.lowerTick),
-                TickMath.getSqrtRatioAtTick(mintParams.upperTick),
-                mintParams.amountA,
-                mintParams.amountB
-            );
 
-            (amount0, amount1) = LiquidityAmounts.getAmountsForLiquidity(
-                TickMath.getSqrtRatioAtTick(newCurrentTick),
-                TickMath.getSqrtRatioAtTick(mintParams.lowerTick),
-                TickMath.getSqrtRatioAtTick(mintParams.upperTick),
-                uint128(liquidity)
-            );
+            if (mintParams.tokenA < mintParams.tokenB) {
+                amount0 = mintParams.amountA;
+                (amount1, liquidity) = calculateSingleAmount(
+                    SingleAmountParams(
+                        amount0,
+                        mintParams.slippage,
+                        true,
+                        TickMath.getSqrtRatioAtTick(newCurrentTick),
+                        TickMath.getSqrtRatioAtTick(mintParams.lowerTick),
+                        TickMath.getSqrtRatioAtTick(mintParams.upperTick)
+                    )
+                );
+            } else {
+                amount1 = mintParams.amountA;
+                (amount0, liquidity) = calculateSingleAmount(
+                    SingleAmountParams(
+                        amount1,
+                        mintParams.slippage,
+                        false,
+                        TickMath.getSqrtRatioAtTick(newCurrentTick),
+                        TickMath.getSqrtRatioAtTick(mintParams.lowerTick),
+                        TickMath.getSqrtRatioAtTick(mintParams.upperTick)
+                    )
+                );
+            }
```

Comment:
@pradyuman-verma You can do it simply way, right? Why code duplicate? Or am I missing something?

Reply:
In one case, the reverseFlag is false. So have to do it twice.